bindings/rust/bench/benches/handshake.rs

```diff
@@ -1,23 +1,36 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-use bench::{S2NHarness, TlsBenchHarness};
+use bench::{RustlsHarness, S2NHarness, TlsBenchHarness};
 use criterion::{criterion_group, criterion_main, BatchSize, Criterion};
 
-pub fn criterion_benchmark(c: &mut Criterion) {
+pub fn bench_handshake(c: &mut Criterion) {
     let mut group = c.benchmark_group("handshake");
 
-    group.bench_function("s2n-tls", |b| {
-        // generate all inputs (s2n-tls objects) before benchmarking handshakes
-        b.iter_batched_ref(
-            || S2NHarness::new(),
-            |s2n_tls| {
-                s2n_tls.handshake().unwrap();
-            },
-            BatchSize::SmallInput,
-        )
-    });
+    macro_rules! bench_handshake_for_libraries {
+        ($(($lib_name:expr, $lib_type:ty),)*) => {
+        $(
+            // generate all inputs (s2n-tls objects) before benchmarking handshakes
+            group.bench_function($lib_name, |b| {
+                b.iter_batched_ref(
+                    || <$lib_type>::default().unwrap(),
+                    |harness| {
+                        harness.handshake().unwrap();
+                    },
+                    BatchSize::SmallInput,
+                )
+            });
+        )*
+        }
+    }
+
+    bench_handshake_for_libraries! {
```

Comment:
Since we're iterating through types of structs that implement a trait, there's not really a good way that I know of of doing that. Macros were the easiest and most concise way to do it, at the expense of some readability.

Reply:
If we're willing to incur a single dynamic dispatch per benchmark run, we could use trait objects and iterate through a Vec of those instead.

Reply:
If we did this, we couldn't use `iter_batched_ref()` to generate the needed structs before timing the handshakes, since the vector would have to exist outside of `bench_function()`. If we make `TlsBenchHarness` structs implement `Clone`, we could clone them as the initialization part of the benchmarking, but `Clone` isn't immediately derivable for either s2n-tls or Rustls. 

If we instead chose to implement say a `reinit()` method that just recycles the internal things the harness uses and resets the connections to pre-handshake, then a vec of trait objects would work, but that adds some overhead to the benchmark that wouldn't be ideal.

Macros still seem to be the best option to me, sacrificing a little readability for concision and performance.